packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts

```diff
@@ -151,6 +151,38 @@ const y: number = x!;
 const x: number | null = null;
 class Foo {
   prop: number = x!;
+}
+    `,
+    `
+class T {
+  a = 'a' as const;
+}
+    `,
+    `
+class T {
+  a = 3 as 3;
+}
+    `,
+    `
+class T {
+  a = 10;
+}
+    `,
```

Comment:
[Tests]: I apologize, I didn't notice this previously, but I think the tests that include no assertions can be removed, as the rule checks specifically for unnecessary type assertions ([there's also this one](https://github.com/typescript-eslint/typescript-eslint/blob/6f2326bea63527e94dff6b2d2ae38f51387d894e/packages/eslint-plugin/tests/rules/no-unnecessary-type-assertion.test.ts#L179-L181)).

Reply:
Thanks! I removed in [a320eaf](https://github.com/typescript-eslint/typescript-eslint/pull/10618/commits/a320eaf6cd280f6d6941eede2c368a0e57b9cd84).